src/main/java/de/tum/in/www1/artemis/web/rest/repository/RepositoryResource.java

```diff
@@ -286,21 +339,56 @@ public RepositoryResource(UserService userService, AuthorizationCheckService aut
 
         Repository repository;
         try {
-            repository = getRepository(domainId);
+            repository = getRepository(domainId, true);
         }
         catch (IllegalAccessException ex) {
             return forbidden();
         }
+        catch (CheckoutConflictException | WrongRepositoryStateException ex) {
+            return new ResponseEntity<>(HttpStatus.CONFLICT);
+        }
+        catch (GitAPIException ex) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
         try {
             repositoryService.commitChanges(repository);
         }
+        catch (CheckoutConflictException | WrongRepositoryStateException ex) {
+            return new ResponseEntity<>(HttpStatus.CONFLICT);
```

Comment:
There is quite a lot duplicated code for error handling now. Can we somehow extract all error handling into a new method so that we get rid of the duplicated code

Reply:
You are right! I'll implement a task executioner that handles the exceptions.